Sources/ImageFetcher/ImageFetcher.swift

```diff
@@ -224,93 +193,72 @@ public extension ImageFetcher {
     /// - Parameter key: The configuation of the image to load.
     /// - Returns:An image instance that has been previously cached. Nil if not found.
     func load(image key: ImageConfiguration) async -> Image? {
-        do {
-            guard let cachedData = try? await cache.data(key.key),
-                  let image = Image(data: cachedData)?.decompressed() else {
-                      return nil
-                  }
+        guard let cachedData = try? await cache.data(key.key) else {
+            return nil
+        }
 
-            return image
+        let decompressTask = Task(priority: key.priority) {
+            try await decompress(cachedData, imageConfiguration: key)
         }
+
+        insertTask(decompressTask, key: key)
+        return try? await decompressTask.value.value
     }
 }
 
 // MARK: - Private Methods
 private extension ImageFetcher {
-    func insertTask(_ imageFetcherTask: ImageFetcherTask) {
-        lock.lock()
-        defer { lock.unlock() }
+    func download(_ imageConfiguration: ImageConfiguration) async throws -> ImageSource {
+        do {
+            let (data, _) = try await networking.load(URLRequest(url: imageConfiguration.url))
```

Comment:
I intend to change the signature of `Networking.load` to `(URLRequest) async throws -> Data` in a future PR